src/main/haskell/language-kore/app/parser/Main.hs

```diff
@@ -77,33 +95,73 @@ main = do
   case localOptions options of
     Nothing -> return () -- global options parsed, but local failed; exit gracefully
     Just KoreParserOptions
-         { fileName
-         , willPrint
-         , willVerify
-         , willChkAttr
-         } -> do
-      parsedDefinition <- mainParse fileName
-      when willVerify $ mainVerify willChkAttr parsedDefinition
-      when willPrint  $ putStrLn (prettyPrintToString parsedDefinition)
-
-
--- | IO action that parses a kore definition from a filename and prints timing information.
-mainParse :: String -> IO KoreDefinition
-mainParse fileName = do
+        { fileName
+        , patternFileName
+        , mainModuleName
+        , willPrint
+        , willVerify
+        , willChkAttr
+        }
+      -> do
+        parsedDefinition <- mainDefinitionParse fileName
+        indexedModules <- if willVerify
+            then mainVerify willChkAttr parsedDefinition
+            else return Map.empty
+        when willPrint $ putStrLn (prettyPrintToString parsedDefinition)
+
+        when (patternFileName /= "") $ do
+            parsedPattern <- mainPatternParse patternFileName
+            when willVerify $ do
+                indexedModule <-
+                    mainModule (ModuleName mainModuleName) indexedModules
+                mainPatternVerify indexedModule parsedPattern
+            when willPrint  $ putStrLn (prettyPrintToString parsedPattern)
```

Comment:
So, when willPrint, it will print both the definition and the pattern?  What if I only want to see one of them?